src/trix/models/document.coffee

```diff
@@ -302,9 +302,12 @@ class Trix.Document extends Trix.Object
     new @constructor @blockList.consolidateFromIndexToIndex(startIndex, endIndex)
 
   getDocumentAtRange: (range) ->
-    range = normalizeRange(range)
-    blocks = @blockList.getSplittableListInRange(range).toArray()
-    new @constructor blocks
+    if @isEmpty()
+      this
```

Comment:
To preserve document immutability, methods should always return a new `Document` instance. Never `this`. Unless you've found a performance issue, I don't think it's worth optimizing here with an `isEmpty` check.